### DISCUSSION TITLE
Using Ant media in nodejs

### DISCUSSION THREAD

**Discussion by Piyush5802** [ant-media/Ant-Media-Server#6574] [Q&A]
Is there any one who has used ant media with nodejs. I m searching for creating api for creating stream using nodejs and i tried something also but getting errors. seems like there is something else i need to setup in the code. If there is anyOne Please help me out

**Comment by lastpeony** [comment]
You can talk to ant media server using nodejs or any other framework in any programming language through REST.
https://antmedia.io/rest/#
Define a REST API jwt token and use it in your nodejs server to talk with ams. 
https://antmedia.io/docs/category/rest-api-guide/

**Comment by rahul7827** [comment]
Hello @Piyush5802 

Do share some code snippet along with error. That will be helpful.

Have you enabled your public IP from management panel ?

**Reply by Piyush5802** [reply]
![Screenshot 2024-08-12 170204](https://github.com/user-attachments/assets/e6af830c-3c75-4503-b2db-e67fdc702e2b)
when i m using api mentioned in ant media documentation i got this error, if you have any piece of code for creating stream and get stream url and other details in response, please share if you have anything

**Reply by rahul7827** [reply]
Hello @Piyush5802 

Sorry, I missed your reply. It seems you have not whitelisted you IP in "ant media console" application. 

1. Login to antmedia console application.
2. Select the application you are using (like: LiveApp, webRTCAppEE).
3. Move to setting tab, In the bottom you will see IP related setting.

Add 0.0.0.0 as IP and click save * (**This is just for testing, This will whitelist all the IP, So it will be publicly exposed**).

Trigger api to create stream, I hope it will work now.

 **Do not forget to revert the IP** in the setting.
